qiskit/circuit/gate.py

```diff
@@ -42,7 +42,6 @@ def __init__(
             params: A list of parameters.
             label: An optional label for the gate.
         """
-        self.definition = None
```

Comment:
Oops. I didn't mean to include this here, but I think it's safe since `Instruction.__init__` sets `self._definition = None` and it saves some measurable time on this by avoiding a call to `@definition.setter`. I'll shove to a separate PR

Reply:
The PR that introduced `ControlledGate` deliberately included this line, which is what makes me particularly nervous about it, but if it passes the tests, I think it ought to be fine.